ranges.md

```diff
@@ -0,0 +1,384 @@
+
+    1. Comparators with the Operator of `>` will include a given SemVer
+       Version if their Version has precedence greater than that of the
+       SemVer Version being considered.
+
+    1. Comparators with the Operator of `>=` will include a given SemVer
+       Version if their Version has precedence greater than or equal to
+       that of the SemVer Version being considered.
+
+    1. Comparators with the Operator of `<` will include a given SemVer
+       Version if their Version has precedence less than that of the
+       SemVer Version being considered.
+
+    1. Comparators with the Operator of `<=` will include a given SemVer
+       Version if their Version has precedence less than or equal to that
+       of the SemVer Version being considered.
+
+1. Inclusion of PRERELEASE idenfiers in Comparator Sets.
+
+    SemVer Versions that include PRERELEASE identifiers are assumed to be
+    unstable and/or incomplete.
+
+    Additionally, the presence of PRERELEASE identifiers in a version such
+    as `I.J.K-PRERELEASE` causes the SemVer Version to have a _lower_
+    precedence than `I.J.K`.  Thus, it could cause a Range such as `<2.0.0`
+    to include the version `2.0.0-rc.0`, which likely contains breaking API
+    changes that the consumer is not prepared to handle.
+
+    To avoid surprising behavior for consumers, the inclusion of SemVer
+    Versions with PRERELEASE identifiers is as follows.
+
+    1. SemVer Versions containing PRERELEASE identifiers MUST NOT be
+       included by a SemVer Comparator Set unless they are included by all
+       of the Comparators in the Comparator Set, and one or more of the
+       following conditions are met:
+
+        1. The implementation has provided the user with an option to
+           explicitly include PRERELEASE versions, and the user has set this
+           option.  This SHOULD NOT be the default behavior when resolving
+           software dependencies.
+
+        1. One or more of the Comparators in the Comparator Set have the
+           same MAJOR, MINOR, and PATCH version as the SemVer Version
+           string being considered, and has a PRERELEASE version, and would
+           normally include the SemVer Version string by virtue of
+           precedence comparison.
+
+            For example, the Range `>=1.0.0-alpha` would include the
+            Version `1.0.0-beta` but _not_ the Version `1.0.1-beta`.
+
+    1. SemVer Version strings without a PRERELEASE version are included in
+       a SemVer Comparator Set if they are included by all of the
+       Comparators in the Comparator Set.
+
+1. `Advanced Comparators` are defined in terms of their reduction to a set
+   of primitive Comparators.
+
+    1. The empty string, asterisk `*`, capital `X` and lowercase `x` are
+       equivalent to `>=0.0.0`, or `>=0.0.0-0` if PRERELEASE versions are
+       being included.
+
+    1. A "tilde version" of the form `~I.J.K` will match any version with a
```

Comment:
calling out this operator is spelled as `~>` in cocoapods / bundler / rubygems

Reply:
Are you sure about RubyGems? According to https://github.com/semver/semver/pull/584#issuecomment-705392992, RubyGems `~>` is something different from this. That comment says "it allows the last given number to increase". Tilde allows specifically K to increase. So for example `~>1.2` means `>=1.2.0 <2.0.0-0` while `~1.2` means `>=1.2.0 <1.3.0-0`.